pyface/ui/qt4/console/console_widget.py

```diff
@@ -122,6 +122,11 @@ class ConsoleWidget(QtGui.QWidget):
         [QtCore.Qt.Key_C, QtCore.Qt.Key_G, QtCore.Qt.Key_O, QtCore.Qt.Key_V]
     )
 
+    #: A list of connected Qt signals to be removed before destruction.
+    #: First item in the tuple is the Qt signal. The second item is the event
+    #: handler.
+    _connections_to_remove = []
```

Comment:
I did not read the scope careful enough. This instantiation needs to live inside `__init__`, otherwise we get one list that is shared among all instances of `ConsoleWidget`.